js/index.js

```diff
@@ -68,35 +37,89 @@ const textarea = document.querySelector('textarea[name="message"]');
 
 //callback function
-function submitButton (e) {
+function submitButton(e) {
   e.preventDefault();
   //display messages in list
- const item = document.createElement('li');
-  item.innerHTML = `<a href="mailto:${email.value}">${name.value}</a><span>${textarea.value}</span>`
-    messageSelection.appendChild(item);
-  const removeButton = document.createElement('button');
-    removeButton.innerHTML = 'remove';
-    removeButton.setAttribute('type', 'button');
-    removeButton.addEventListener('click', deleteItem) 
-    item.appendChild(removeButton);
+  const item = document.createElement("li");
+  item.innerHTML = `<a href="mailto:${email.value}">${name.value}</a><span>${textarea.value}</span>`;
+  messageSelection.appendChild(item);
+  const removeButton = document.createElement("button");
+  removeButton.innerHTML = "remove";
+  removeButton.setAttribute("type", "button");
+  removeButton.addEventListener("click", deleteItem);
+  item.appendChild(removeButton);
 
   //log the values of e, name, email, and text area
   console.log(e);
   console.log(name.value);
   console.log(email.value);
   console.log(textarea.value);
-  
+
   //clears the form
   messageForm.reset();
 }
 
 //create the deleteItem function
-function deleteItem (e) {
+function deleteItem(e) {
   e.target.parentNode.remove();
 }
+// }
+//use Fetch API to get same GitHub API url
+const githubAPI = "https://api.github.com/users/jenhealthy/repos";
+
+//stretch goals: Chain a catch() function to your fetch call to handle errors from the server
+fetch(githubAPI)
+  .then((response) => {
+    if (response.status != 200) {
+      throw new Error(`There was a problem with your fetching.`);
+    }
+    //Chain a then method to your fetch call and pass it a function that returns the response JSON data
+    return response.json();
+  })
+  //then method and pass it a function, inside of which you can paste the code from your previous "load" event listener function
+  .then(githubRequest)
+  .catch((err) => {
+    console.error({ err: err });
+  });
+
+function githubRequest(repositories) {
+  console.log({ respositories: repositories });
+
+  //handle response from server
+  // githubRequest.addEventListener("load", gitHandler);
 
+  //serverside you parse the data into object
+  // const repositories = JSON.parse(this.response);
 
+  const projectList = document.querySelector("#projects");
+  const ulElement = projectList.querySelector("ul");
+  for (let i = 0; i < repositories.length; i++) {
+    const repo = repositories[i];
+    let project = document.createElement("li");
+    project.innerHTML = `<a href= ${repositories[i].html_url}>${repositories[i].name}</a>`;
+    ulElement.appendChild(project);
+    let newList = document.createElement("ul");
+    project.appendChild(newList);
+    let projectDescription = document.createElement("li");
+    projectDescription.innerText = repositories[i].description;
+    newList.appendChild(projectDescription);
```

Comment:
You also may add a condition in case you will want to avoid adding `li` with empty description string. For example,
`repo.description ? newList.appendChild(projectDescription):"" ;`